purchase_request_procurement/models/purchase_request.py

```diff
@@ -0,0 +1,29 @@
+# -*- coding: utf-8 -*-
+# Copyright 2017 Eficent Business and IT Consulting Services S.L.
+# License LGPL-3.0 or later (http://www.gnu.org/licenses/lgpl-3.0).
+
+from openerp import api, models, _
+from openerp.exceptions import UserError
+
+
+class PurchaseRequest(models.Model):
+    _inherit = 'purchase.request'
+
+    @api.multi
+    def button_rejected(self):
+        if any([s == 'done' for s in self.mapped(
+                'line_ids.procurement_id.state')]):
+            raise UserError(
+                _('You cannot reject a purchase request with lines related to '
+                  'done procurement orders.'))
+        return super(PurchaseRequest, self).button_rejected()
+
+    @api.multi
+    def button_draft(self):
+        if any([s == 'done' for s in self.mapped(
+                'line_ids.procurement_id.state')]):
+            raise UserError(
+                _('You cannot set back to draft a purchase request with lines '
+                  'related to done procurement orders.'))
```

Comment:
Yes indeed. Then maybe we should have two methods which have the condition and two other methods that check the condition and raise an error. As you wish. But I think it would be better to at least, move thoses check in a specific method.